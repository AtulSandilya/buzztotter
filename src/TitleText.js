import { Text, View } from 'react-native';
import React, { Component, PropTypes } from 'react';

import {styles} from './Styles.js'

export default class TitleText extends Component {
  static propTypes = {
    title: React.PropTypes.string.isRequired,
  }

  render() {
    return(
      <View style={styles.titleTextContainer}>
        <Text style={styles.titleText}>
          {this.props.title}
        </Text>
      </View>
    );
  }
}
