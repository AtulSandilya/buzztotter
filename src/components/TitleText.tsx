import * as React from "react";
import { Component } from 'react';
import { Text, View } from 'react-native';

import {globalStyles} from './GlobalStyles';

interface TitleTextProps {
  title: string;
}

const TitleText: React.StatelessComponent<TitleTextProps> = ({title}) => (
  <View style={globalStyles.titleTextContainer}>
    <Text style={globalStyles.titleText}>
      {title}
    </Text>
  </View>
)

export default TitleText;
