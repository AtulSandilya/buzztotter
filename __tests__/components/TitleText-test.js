import 'react-native';
import React from 'react';
import TitleText from '../../src/components/TitleText';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <TitleText
      title="Title:"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
