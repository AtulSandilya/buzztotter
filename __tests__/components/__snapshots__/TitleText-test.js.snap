exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "borderBottomWidth": 1,
      "borderColor": "#dddddd",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 20,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#8ED0BA",
        "fontSize": 30,
        "fontWeight": "300",
        "paddingBottom": 10,
      }
    }>
    Title:
  </Text>
</View>
`;
